lib/SILGen/SILGenPattern.cpp

```diff
@@ -1031,7 +1031,8 @@ void PatternMatchEmission::emitDispatch(ClauseMatrix &clauses, ArgArray args,
         } else {
           Loc = clauses[firstRow].getCasePattern()->getStartLoc();
         }
-        SGF.SGM.diagnose(Loc, diag::unreachable_case, isDefault);
+        if (!isDefault)
+          SGF.SGM.diagnose(Loc, diag::unreachable_case);
```

Comment:
@jckarter Is this reasonable?

Reply:
I don't recall the history here—why are we diagnosing this in SILGen?

Reply:
Just checking, since I've been out of it: we do diagnose unreachable defaults in Sema when we can, right?

Reply:
As of now: no (and I don’t believe we should because of things like [SR-3278](https://bugs.swift.org/browse/SR-3278)).  `default` early-exits the current exhaustiveness algorithm.

Reply:
Ah, right. I guess that's on me to put in once we switch to resilient builds with the exhaustive/non-exhaustive distinction on.